// @steera/form-manager/src/Comp/FormRenderer/RenderComponents/RadioBox/RadioBox.utils.ts
import {FormRendererProps} from '../../FormRenderer.interface';
import {ComponentOptionItemType as OptionType} from '../../../Comp.interface';

export function getID(option: OptionType) {
  return option.input.value;
}

export function getLabel(option: OptionType) {
  return option.input.value;
}

export function getName(option: OptionType) {
  return option.input.value;
}

export function getValue(option: OptionType) {
  return option.input.value;
}

export function getDisabled(option: OptionType) {
  return option.input.checked;
}

export function getRequired(option: OptionType) {
  return option.input.checked;
}

export function getChecked(
  templateValue: FormRendererProps['templateValues'],
  name: string
) {
  return (value: string) => {
    if (templateValue && templateValue.hasOwnProperty(name)) {
      return templateValue[name] === value;
    }
    return false;
  };
}

export function getClassName(option: OptionType) {
  return option.input.value;
}

export function getInvalidMessage(option: OptionType) {
  return option.input.value;
}
