// @steera/form-manager/dist/cjs/Comp/FormGenerator/Store/Store.initialstate.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.initialState = void 0;
const Store_constants_1 = require("./Store.constants");
const pathUtils_1 = require("../../../Utils/pathUtils");
const RootElementPath = pathUtils_1.Path.makePathNode(Store_constants_1.TemplateComponent.box.pathId, 0);
const RootChildPath = pathUtils_1.Path.addNodeToPath(RootElementPath, pathUtils_1.Path.makePathNode(Store_constants_1.TemplateComponent.addnew.pathId, 0));
const starterElement = {
    title: Store_constants_1.TemplateComponent.box.title,
    path: RootElementPath,
    pathId: Store_constants_1.TemplateComponent.box.pathId,
    options: Store_constants_1.TemplateComponent.box.options,
    children: [
        {
            title: Store_constants_1.TemplateComponent.addnew.title,
            path: RootChildPath,
            pathId: Store_constants_1.TemplateComponent.addnew.pathId,
            options: Store_constants_1.TemplateComponent.addnew.options,
            children: [],
        },
    ],
};
exports.initialState = {
    isPreview: false,
    templateId: '',
    templateName: '',
    selectedNodePath: '',
    template: [starterElement],
};
/**
 * An template-canvas at all conditions should have a FLEX_COLUMN
 * component as the starter. This ensure a consistant starting point
 * for the template object
 *
 */
