// @steera/form-manager/dist/cjs/Comp/FormGenerator/Store/Store.actions.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.deleteNode = exports.closeConfigWindow = exports.updateOptionItem = exports.selectComponent = exports.togglePreview = exports.addToCanvas = exports.onChangeTemplateName = exports.closeLeftSection = exports.addNewItem = exports.propsChange = void 0;
const Store_constants_1 = require("./Store.constants");
function propsChange(payload) {
    return {
        type: Store_constants_1.ACTION.PROPS_CHANGE,
        payload,
    };
}
exports.propsChange = propsChange;
function addNewItem(payload) {
    return {
        type: Store_constants_1.ACTION.ADD_NEW,
        payload,
    };
}
exports.addNewItem = addNewItem;
function closeLeftSection() {
    return {
        type: Store_constants_1.ACTION.CLOSE_SECTION,
    };
}
exports.closeLeftSection = closeLeftSection;
function onChangeTemplateName(payload) {
    return {
        type: Store_constants_1.ACTION.CHANGE_NAME,
        payload,
    };
}
exports.onChangeTemplateName = onChangeTemplateName;
/** New & Resused Actions */
function addToCanvas(payload) {
    return {
        type: Store_constants_1.ACTION.ADD_TO_CANVAS,
        payload,
    };
}
exports.addToCanvas = addToCanvas;
function togglePreview() {
    return {
        type: Store_constants_1.ACTION.TOGGLE_PREVIEW,
    };
}
exports.togglePreview = togglePreview;
function selectComponent(payload) {
    return {
        type: Store_constants_1.ACTION.SELECT_COMPONENT,
        payload,
    };
}
exports.selectComponent = selectComponent;
function updateOptionItem(payload) {
    return {
        type: Store_constants_1.ACTION.UPDATE_OPTION,
        payload,
    };
}
exports.updateOptionItem = updateOptionItem;
function closeConfigWindow() {
    return {
        type: Store_constants_1.ACTION.CLOSE_SECTION,
    };
}
exports.closeConfigWindow = closeConfigWindow;
function deleteNode() {
    return {
        type: Store_constants_1.ACTION.DELETE_NODE,
        payload: null,
    };
}
exports.deleteNode = deleteNode;
