// @steera/form-manager/src/Comp/FormRenderer/RenderComponents/DateBox/DateBox.utils.ts
import {FormRendererProps} from '../../FormRenderer.interface';
import {ComponentOptionItemType as OptionType} from '../../../Comp.interface';

export function getID(option: OptionType) {
  return option.input.value;
}

export function getLabel(option: OptionType) {
  return option.input.value;
}

export function getName(option: OptionType) {
  return option.input.value;
}

export function getDisabled(option: OptionType) {
  return option.input.checked;
}

export function getRequired(option: OptionType) {
  return option.input.checked;
}

export function getClassName(option: OptionType) {
  return option.input.value;
}

export function getMax(option: OptionType) {
  if (option.input.value.trim() === '') return undefined;
  else return option.input.value;
}

export function getMin(option: OptionType) {
  if (option.input.value.trim() === '') return undefined;
  else return option.input.value;
}

export function getInvalidMessage(option: OptionType) {
  return option.input.value;
}

export function getValue(
  templateValues: FormRendererProps['templateValues'],
  name: string
) {
  if (templateValues && templateValues.hasOwnProperty(name)) {
    return templateValues[name];
  }
}
