// @steera/form-manager/dist/esm/Comp/FormGenerator/Store/Store.actions.js
import { ACTION } from './Store.constants';
export function propsChange(payload) {
    return {
        type: ACTION.PROPS_CHANGE,
        payload,
    };
}
export function addNewItem(payload) {
    return {
        type: ACTION.ADD_NEW,
        payload,
    };
}
export function closeLeftSection() {
    return {
        type: ACTION.CLOSE_SECTION,
    };
}
export function onChangeTemplateName(payload) {
    return {
        type: ACTION.CHANGE_NAME,
        payload,
    };
}
/** New & Resused Actions */
export function addToCanvas(payload) {
    return {
        type: ACTION.ADD_TO_CANVAS,
        payload,
    };
}
export function togglePreview() {
    return {
        type: ACTION.TOGGLE_PREVIEW,
    };
}
export function selectComponent(payload) {
    return {
        type: ACTION.SELECT_COMPONENT,
        payload,
    };
}
export function updateOptionItem(payload) {
    return {
        type: ACTION.UPDATE_OPTION,
        payload,
    };
}
export function closeConfigWindow() {
    return {
        type: ACTION.CLOSE_SECTION,
    };
}
export function deleteNode() {
    return {
        type: ACTION.DELETE_NODE,
        payload: null,
    };
}
