// @steera/form-manager/src/Comp/FormGenerator/RenderComponents/Fieldset/Fieldset.tsx
import React from 'react';
import {Fieldset as F} from '../../CanvasComponents';
import {
  getLabel,
  getFlexDirection,
  getMainClass,
  getLegendClass,
  getPaddingTop,
  getPaddingRight,
  getPaddingBottom,
  getPaddingLeft,
} from './Fieldset.utils';

export function Fieldset(props: any) {
  const value = React.useMemo(() => {
    const paddingTop = getPaddingTop(props.options.paddingTop);
    return {
      style: {
        paddingTop,
        flexDirection: getFlexDirection(props.options.flexDirection),
        paddingRight: getPaddingRight(props.options.paddingRight),
        paddingBottom: getPaddingBottom(props.options.paddingBottom),
        paddingLeft: getPaddingLeft(props.options.paddingLeft),
      },
      title: getLabel(props.options.legend),
      mainClass: getMainClass(props.options.mainClass),
      legendClass: getLegendClass(props.options.legendClass),
      legendStyle: {top: paddingTop},
    };
  }, [props.options]);

  return (
    <F
      pathId={props.pathId}
      path={props.path}
      style={value.style}
      title={value.title}
      mainClass={value.mainClass}
      legendStyle={value.legendStyle}
      legendClass={value.legendClass}>
      {props.children}
    </F>
  );
}
