// @steera/form-manager/src/Comp/FormGenerator/Store/Store.constants.ts
import {ComponentConfig} from '../../Comp.config';

export const ACTION = {
  ADD_NEW: 'formgenerator/addnew',
  ADD_COMPONENT: 'formgenerator/addcomponent',
  DELETE_NODE: 'formgenerator/deletenode',
  PROPS_CHANGE: 'formgenerator/propschange',
  CHANGE_NAME: 'formgenerator/changetemplatename',
  /**New and Reused Actions */

  TOGGLE_PREVIEW: 'formgenerator/togglepreview',
  ADD_TO_CANVAS: 'formgenerator/addtocanvas',
  CLOSE_SECTION: 'formgenerator/configuresection',
  UPDATE_OPTION: 'formgenerator/updateoptionitem',
  SELECT_COMPONENT: 'formgenerator/selectcomponent',
};
export const TemplateComponent = ComponentConfig;

/* eslint-disable */
// export const TemplateComponent: TemplateComponentType = {
//   [ComponentPath.addnew]: {
//     title: 'Add New',
//     pathKey: ComponentPath.addnew,
//     type: ComponentType.ADD_NEW,
//     options: {},
//   },
//   [ComponentPath.flexrow]: {
//     title: 'Flex Row',
//     pathKey: ComponentPath.flexrow,
//     type: ComponentType.FLEX_ROW,
//     options: {
//       spacingTop: {
//         label: 'Space Top',
//         value: 8,
//         type: 'number',
//         required: true,
//       },
//       spacingRight: {
//         label: 'Space Right',
//         value: 8,
//         type: 'number',
//         required: true,
//       },
//       spacingBottom: {
//         label: 'Space Bottom',
//         value: 8,
//         type: 'number',
//         required: true,
//       },
//       spacingLeft: {
//         label: 'Space Left',
//         value: 8,
//         type: 'number',
//         required: true,
//       },
//       style: {
//         label: 'Element Style',
//         value: {},
//         type: 'key-value-pair',
//         required: false,
//       },
//     },
//   },
//   [ComponentPath.flexcolumn]: {
//     title: 'Flex Column',
//     pathKey: ComponentPath.flexcolumn,
//     type: ComponentType.FLEX_COLUMN,
//     options: {
//       spacingTop: {
//         label: 'Space Top',
//         value: 8,
//         type: 'number',
//         required: true,
//       },
//       spacingRight: {
//         label: 'Space Right',
//         value: 8,
//         type: 'number',
//         required: true,
//       },
//       spacingBottom: {
//         label: 'Space Bottom',
//         value: 8,
//         type: 'number',
//         required: true,
//       },
//       spacingLeft: {
//         label: 'Space Left',
//         value: 8,
//         type: 'number',
//         required: true,
//       },
//       style: {
//         label: 'Element Style',
//         value: {},
//         type: 'key-value-pair',
//         required: false,
//       },
//     },
//   },
//   [ComponentPath.emailbox]: {
//     title: 'Email Block',
//     pathKey: ComponentPath.emailbox,
//     type: ComponentType.EMAIL_BOX,
//     options: {
//       label: {
//         label: 'Input Label',
//         value: 'Default Label',
//         type: 'string',
//         required: true,
//       },
//       name: {
//         label: 'Input Name',
//         value: 'emailboxname',
//         type: 'string',
//         required: true,
//       },
//       value: {
//         label: 'Input value',
//         value: '',
//         type: 'string',
//         required: false,
//       },
//       disabled: {
//         label: 'Disabled',
//         value: false,
//         type: 'boolean',
//         required: false,
//       },
//       required: {
//         label: 'Required',
//         value: false,
//         type: 'boolean',
//         required: true,
//       },
//       placeholder: {
//         label: 'Input Placeholder',
//         value: 'default placeholder',
//         type: 'string',
//         required: false,
//       },
//       pattern: {
//         label: 'Input Pattern',
//         value: '',
//         type: 'string',
//         required: true,
//       },
//       minLength: {
//         label: 'Min Length for Value',
//         value: null,
//         type: 'number',
//         required: false,
//       },
//       maxLength: {
//         label: 'Max Length for Value',
//         value: null,
//         type: 'number',
//         required: false,
//       },
//     },
//   },
//   [ComponentPath.passwordbox]: {
//     title: 'Password Block',
//     pathKey: ComponentPath.passwordbox,
//     type: ComponentType.PASSWORD_BOX,
//     options: {
//       label: {
//         label: 'Input Label',
//         value: 'Default Label',
//         type: 'string',
//         required: true,
//       },
//       name: {
//         label: 'Input Name',
//         value: 'passwordboxname',
//         type: 'string',
//         required: true,
//       },
//       value: {
//         label: 'Input value',
//         value: '',
//         type: 'string',
//         required: false,
//       },
//       disabled: {
//         label: 'Disabled',
//         value: false,
//         type: 'boolean',
//         required: false,
//       },
//       required: {
//         label: 'Required',
//         value: false,
//         type: 'boolean',
//         required: true,
//       },
//       placeholder: {
//         label: 'Input Placeholder',
//         value: 'passwordbox placeholder',
//         type: 'string',
//         required: false,
//       },
//       pattern: {
//         label: 'Input Pattern',
//         value: '',
//         type: 'string',
//         required: true,
//       },
//       minLength: {
//         label: 'Min Length for Value',
//         value: null,
//         type: 'number',
//         required: false,
//       },
//       maxLength: {
//         label: 'Max Length for Value',
//         value: null,
//         type: 'number',
//         required: false,
//       },
//     },
//   },
//   [ComponentPath.numberbox]: {
//     title: 'Number Block',
//     pathKey: ComponentPath.numberbox,
//     type: ComponentType.NUMBER_BOX,
//     options: {
//       label: {
//         label: 'Input Label',
//         value: 'Default Label',
//         type: 'string',
//         required: true,
//       },
//       name: {
//         label: 'Input Name',
//         value: 'numberboxname',
//         type: 'string',
//         required: true,
//       },
//       value: {
//         label: 'Input value',
//         value: '',
//         type: 'string',
//         required: false,
//       },
//       disabled: {
//         label: 'Disabled',
//         value: false,
//         type: 'boolean',
//         required: false,
//       },
//       required: {
//         label: 'Required',
//         value: false,
//         type: 'boolean',
//         required: true,
//       },
//       placeholder: {
//         label: 'Input Placeholder',
//         value: 'numberbox placeholder',
//         type: 'string',
//         required: false,
//       },
//       pattern: {
//         label: 'Input Pattern',
//         value: '',
//         type: 'string',
//         required: true,
//       },
//       minLength: {
//         label: 'Min Length for Value',
//         value: null,
//         type: 'number',
//         required: false,
//       },
//       maxLength: {
//         label: 'Max Length for Value',
//         value: null,
//         type: 'number',
//         required: false,
//       },
//     },
//   },
//   [ComponentPath.textbox]: {
//     title: 'Textbox Block',
//     pathKey: ComponentPath.textbox,
//     type: ComponentType.TEXT_BOX,
//     options: {
//       label: {
//         label: 'Input Label',
//         value: 'Default Label',
//         type: 'string',
//         required: true,
//       },
//       name: {
//         label: 'Input Name',
//         value: 'textboxname',
//         type: 'string',
//         required: true,
//       },
//       value: {
//         label: 'Input value',
//         value: '',
//         type: 'string',
//         required: false,
//       },
//       disabled: {
//         label: 'Disabled',
//         value: false,
//         type: 'boolean',
//         required: false,
//       },
//       required: {
//         label: 'Required',
//         value: false,
//         type: 'boolean',
//         required: true,
//       },
//       placeholder: {
//         label: 'Input Placeholder',
//         value: 'textbox placeholder',
//         type: 'string',
//         required: false,
//       },
//       pattern: {
//         label: 'Input Pattern',
//         value: '',
//         type: 'string',
//         required: true,
//       },
//       minLength: {
//         label: 'Min Length for Value',
//         value: null,
//         type: 'number',
//         required: false,
//       },
//       maxLength: {
//         label: 'Max Length for Value',
//         value: null,
//         type: 'number',
//         required: false,
//       },
//     },
//   },
// };
