// @steera/form-manager/src/Comp/FormGenerator/Store/Store.initialstate.ts
import {TemplateState} from './Store.interface';
import {TemplateComponent} from './Store.constants';
import {Path} from '../../../Utils/pathUtils';
import {TemplateNodeType} from '../../Comp.interface';

const RootElementPath = Path.makePathNode(TemplateComponent.box.pathId, 0);
const RootChildPath = Path.addNodeToPath(
  RootElementPath,
  Path.makePathNode(TemplateComponent.addnew.pathId, 0)
);

const starterElement: TemplateNodeType = {
  title: TemplateComponent.box.title,
  path: RootElementPath,
  pathId: TemplateComponent.box.pathId,
  options: TemplateComponent.box.options,
  children: [
    {
      title: TemplateComponent.addnew.title,
      path: RootChildPath,
      pathId: TemplateComponent.addnew.pathId,
      options: TemplateComponent.addnew.options,
      children: [],
    },
  ],
};

export const initialState: TemplateState = {
  isPreview: false,
  templateId: '',
  templateName: '',
  selectedNodePath: '',
  template: [starterElement],
};

/**
 * An template-canvas at all conditions should have a FLEX_COLUMN
 * component as the starter. This ensure a consistant starting point
 * for the template object
 *
 */
