// @steera/form-manager/dist/cjs/Comp/FormRenderer/RenderComponents/Title/Title.utils.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getPaddingLeft = exports.getPaddingBottom = exports.getPaddingRight = exports.getPaddingTop = exports.getClassName = exports.getValue = exports.getLabel = exports.getID = void 0;
function getID(option) {
    return option.input.value;
}
exports.getID = getID;
function getLabel(option) {
    return option.input.value;
}
exports.getLabel = getLabel;
function getValue(option) {
    return option.input.value;
}
exports.getValue = getValue;
function getClassName(option) {
    return option.input.value;
}
exports.getClassName = getClassName;
function getPaddingTop(option) {
    return `${option.input.value}${option.attribute.unit}`;
}
exports.getPaddingTop = getPaddingTop;
function getPaddingRight(option) {
    return `${option.input.value}${option.attribute.unit}`;
}
exports.getPaddingRight = getPaddingRight;
function getPaddingBottom(option) {
    return `${option.input.value}${option.attribute.unit}`;
}
exports.getPaddingBottom = getPaddingBottom;
function getPaddingLeft(option) {
    return `${option.input.value}${option.attribute.unit}`;
}
exports.getPaddingLeft = getPaddingLeft;
