// @steera/form-manager/dist/cjs/Utils/commonInterface.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.FormRendererComponentTypes = exports.FormGeneratorComponentTypes = exports.Layout = void 0;
var Layout;
(function (Layout) {
    Layout["row"] = "row";
    Layout["colunn"] = "column";
})(Layout = exports.Layout || (exports.Layout = {}));
/**
 * Different Types of Components Handled by FormGenerator
 */
var FormGeneratorComponentTypes;
(function (FormGeneratorComponentTypes) {
    FormGeneratorComponentTypes["ADD_NEW"] = "ADD_NEW";
    FormGeneratorComponentTypes["FLEX_ROW"] = "FLEX_ROW";
    FormGeneratorComponentTypes["EMAIL_BOX"] = "EMAIL_BOX";
    FormGeneratorComponentTypes["FLEX_COLUMN"] = "FLEX_COLUMN";
    FormGeneratorComponentTypes["PASSWORD_BOX"] = "PASSWORD_BOX";
    FormGeneratorComponentTypes["NUMBER_BOX"] = "NUMBER_BOX";
    FormGeneratorComponentTypes["TEXT_BOX"] = "TEXT_BOX";
})(FormGeneratorComponentTypes = exports.FormGeneratorComponentTypes || (exports.FormGeneratorComponentTypes = {}));
/**
 * Different Types of Components Handled by FormRenderer
 */
var FormRendererComponentTypes;
(function (FormRendererComponentTypes) {
    FormRendererComponentTypes["FLEX_ROW"] = "FLEX_ROW";
    FormRendererComponentTypes["EMAIL_BOX"] = "EMAIL_BOX";
    FormRendererComponentTypes["FLEX_COLUMN"] = "FLEX_COLUMN";
    FormRendererComponentTypes["PASSWORD_BOX"] = "PASSWORD_BOX";
    FormRendererComponentTypes["NUMBER_BOX"] = "NUMBER_BOX";
    FormRendererComponentTypes["TEXT_BOX"] = "TEXT_BOX";
})(FormRendererComponentTypes = exports.FormRendererComponentTypes || (exports.FormRendererComponentTypes = {}));
