// @steera/form-manager/dist/cjs/Comp/FormRenderer/RenderComponents/NumberBox/NumberBox.utils.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getValue = exports.getInvalidMessage = exports.getMin = exports.getMax = exports.getClassName = exports.getPlaceholder = exports.getRequired = exports.getDisabled = exports.getDefaultValue = exports.getName = exports.getLabel = exports.getID = void 0;
function getID(option) {
    return option.input.value;
}
exports.getID = getID;
function getLabel(option) {
    return option.input.value;
}
exports.getLabel = getLabel;
function getName(option) {
    return option.input.value;
}
exports.getName = getName;
function getDefaultValue(option) {
    return option.input.value;
}
exports.getDefaultValue = getDefaultValue;
function getDisabled(option) {
    return option.input.checked;
}
exports.getDisabled = getDisabled;
function getRequired(option) {
    return option.input.checked;
}
exports.getRequired = getRequired;
function getPlaceholder(option) {
    return option.input.value;
}
exports.getPlaceholder = getPlaceholder;
function getClassName(option) {
    return option.input.value;
}
exports.getClassName = getClassName;
function getMax(option) {
    return option.input.value;
}
exports.getMax = getMax;
function getMin(option) {
    return option.input.value;
}
exports.getMin = getMin;
function getInvalidMessage(option) {
    return option.input.value;
}
exports.getInvalidMessage = getInvalidMessage;
function getValue(templateValues, name) {
    if (templateValues && templateValues.hasOwnProperty(name)) {
        return templateValues[name];
    }
}
exports.getValue = getValue;
