// @steera/form-manager/dist/cjs/Comp/FormRenderer/RenderComponents/Box/Box.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Box = void 0;
const jsx_runtime_1 = require("react/jsx-runtime");
const react_1 = __importDefault(require("react"));
const CanvasComponents_1 = require("../../CanvasComponents");
const Box_utils_1 = require("./Box.utils");
function Box(props) {
    const style = react_1.default.useMemo(() => ({
        flexDirection: (0, Box_utils_1.getFlexDirection)(props.options.flexDirection),
        paddingTop: (0, Box_utils_1.getPaddingTop)(props.options.paddingTop),
        paddingRight: (0, Box_utils_1.getPaddingRight)(props.options.paddingRight),
        paddingBottom: (0, Box_utils_1.getPaddingBottom)(props.options.paddingBottom),
        paddingLeft: (0, Box_utils_1.getPaddingLeft)(props.options.paddingLeft),
    }), [props.options]);
    return (0, jsx_runtime_1.jsx)(CanvasComponents_1.Box, Object.assign({ style: style }, { children: props.children }));
}
exports.Box = Box;
