// @steera/form-manager/dist/cjs/Comp/FormRenderer/RenderComponents/Fieldset/Fieldset.utils.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getPaddingLeft = exports.getPaddingBottom = exports.getPaddingRight = exports.getPaddingTop = exports.getLegendClass = exports.getMainClass = exports.getFlexDirection = exports.getLabel = void 0;
function getLabel(option) {
    return option.input.value;
}
exports.getLabel = getLabel;
function getFlexDirection(option) {
    return option.input.value;
}
exports.getFlexDirection = getFlexDirection;
function getMainClass(option) {
    return option.input.value;
}
exports.getMainClass = getMainClass;
function getLegendClass(option) {
    return option.input.value;
}
exports.getLegendClass = getLegendClass;
function getPaddingTop(option) {
    return option.input.value;
}
exports.getPaddingTop = getPaddingTop;
function getPaddingRight(option) {
    return option.input.value;
}
exports.getPaddingRight = getPaddingRight;
function getPaddingBottom(option) {
    return option.input.value;
}
exports.getPaddingBottom = getPaddingBottom;
function getPaddingLeft(option) {
    return option.input.value;
}
exports.getPaddingLeft = getPaddingLeft;
