// @steera/form-manager/dist/cjs/Comp/FormRenderer/RenderComponents/Fieldset/Fieldset.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Fieldset = void 0;
const jsx_runtime_1 = require("react/jsx-runtime");
const react_1 = __importDefault(require("react"));
const CanvasComponents_1 = require("../../CanvasComponents");
const Fieldset_utils_1 = require("./Fieldset.utils");
function Fieldset(props) {
    const value = react_1.default.useMemo(() => {
        const paddingTop = (0, Fieldset_utils_1.getPaddingTop)(props.options.paddingTop);
        return {
            style: {
                paddingTop,
                flexDirection: (0, Fieldset_utils_1.getFlexDirection)(props.options.flexDirection),
                paddingRight: (0, Fieldset_utils_1.getPaddingRight)(props.options.paddingRight),
                paddingBottom: (0, Fieldset_utils_1.getPaddingBottom)(props.options.paddingBottom),
                paddingLeft: (0, Fieldset_utils_1.getPaddingLeft)(props.options.paddingLeft),
            },
            title: (0, Fieldset_utils_1.getLabel)(props.options.legend),
            mainClass: (0, Fieldset_utils_1.getMainClass)(props.options.mainClass),
            legendClass: (0, Fieldset_utils_1.getLegendClass)(props.options.legendClass),
            legendStyle: { top: paddingTop },
        };
    }, [props.options]);
    return ((0, jsx_runtime_1.jsx)(CanvasComponents_1.Fieldset, Object.assign({ style: value.style, title: value.title, mainClass: value.mainClass, legendStyle: value.legendStyle, legendClass: value.legendClass }, { children: props.children })));
}
exports.Fieldset = Fieldset;
