// @steera/form-manager/src/Comp/Comp.config.ts
/* eslint-disable */
import {ComponentPath, ComponentType} from './Comp.interface';

export const ComponentConfig: Record<ComponentPath, ComponentType> = {
  [ComponentPath.box]: {
    title: 'Box',
    pathId: ComponentPath.box,
    options: {
      paddingTop: {
        title: 'Spacing Top',
        input: {
          type: 'number',
          value: 8,
          required: true,
          disabled: false,
          min: 0,
          max: 32,
        },
        attribute: {
          unit: 'px',
        },
      },
      paddingBottom: {
        title: 'Spacing Bottom',
        input: {
          type: 'number',
          value: 8,
          required: true,
          disabled: false,
          min: 0,
          max: 32,
        },
        attribute: {
          unit: 'px',
        },
      },
      paddingLeft: {
        title: 'Spacing Left',
        input: {
          type: 'number',
          value: 8,
          required: true,
          disabled: false,
          min: 0,
          max: 32,
        },
        attribute: {
          unit: 'px',
        },
      },
      paddingRight: {
        title: 'Spacing Right',
        input: {
          type: 'number',
          value: 8,
          required: true,
          disabled: false,
          min: 0,
          max: 32,
        },
        attribute: {
          unit: 'px',
        },
      },
      flexDirection: {
        title: 'Orientation',
        input: {
          type: 'select',
          value: 'row',
          options: [
            {label: 'Row', value: 'row'},
            {label: 'Column', value: 'column'},
          ],
          required: true,
          disabled: false,
        },
        attribute: {},
      },
    },
  },
  [ComponentPath.textbox]: {
    title: 'Text Box',
    pathId: ComponentPath.textbox,
    options: {
      id: {
        title: 'Input ID(#)',
        input: {
          type: 'text',
          value: 'ID',
          required: true,
          disabled: false,
          maxLength: 100,
          minLength: 1,
        },
        attribute: {},
      },
      className: {
        title: 'Custom Class Name',
        input: {
          type: 'text',
          value: '',
          required: false,
          disabled: false,
          maxLength: 100,
          minLength: 1,
        },
        attribute: {
          type: 'string',
          name: 'className',
        },
      },
      label: {
        title: 'Title',
        input: {
          type: 'text',
          value: 'label',
          required: true,
          disabled: false,
          maxLength: 100,
          minLength: 1,
        },
        attribute: {},
      },
      name: {
        title: 'Name',
        input: {
          type: 'text',
          value: 'name',
          required: true,
          disabled: false,
          maxLength: 100,
          minLength: 1,
        },
        attribute: {},
      },
      disabled: {
        title: 'Disabled',
        input: {
          type: 'checkbox',
          checked: false,
          name: 'isDisabled',
          value: 'disabled',
        },
        attribute: {},
      },
      required: {
        title: 'Required',
        input: {
          type: 'checkbox',
          checked: false,
          name: 'isRequired',
          value: 'required',
        },
        attribute: {},
      },
      placeholder: {
        title: 'Placeholder',
        input: {
          type: 'text',
          value: 'placeholder',
          required: true,
          disabled: false,
          maxLength: 1200,
          minLength: 1,
        },
        attribute: {},
      },
      pattern: {
        title: 'Pattern',
        input: {
          type: 'text',
          value: '',
          required: true,
          disabled: false,
          maxLength: 1200,
          minLength: 1,
        },
        attribute: {},
      },
      maxLength: {
        title: 'Max Length',
        input: {
          type: 'number',
          value: 0,
          required: false,
          disabled: false,
          min: 0,
        },
        attribute: {},
      },
      minLength: {
        title: 'Min Length',
        input: {
          type: 'number',
          value: 0,
          required: false,
          disabled: false,
          min: 0,
        },
        attribute: {},
      },
      invalidMessage: {
        title: 'Validation Message',
        input: {
          type: 'text',
          value: 'Current value is invalid',
          required: true,
          disabled: false,
          maxLength: 300,
          minLength: 1,
        },
        attribute: {},
      },
    },
  },
  [ComponentPath.addnew]: {
    title: 'Add New',
    pathId: ComponentPath.addnew,
    options: {},
  },
  [ComponentPath.fieldset]: {
    title: 'Input Group',
    pathId: ComponentPath.fieldset,
    options: {
      legend: {
        title: 'Title',
        input: {
          type: 'text',
          value: 'Title',
          required: true,
          disabled: false,
          minLength: 1,
          maxLength: 30,
        },
        attribute: {},
      },
      flexDirection: {
        title: 'Orientation',
        input: {
          type: 'select',
          value: 'row',
          options: [
            {label: 'Row', value: 'row'},
            {label: 'Column', value: 'column'},
          ],
          required: true,
          disabled: false,
        },
        attribute: {},
      },
      mainClass: {
        title: 'ClassName',
        input: {
          type: 'text',
          value: '',
          required: true,
          disabled: false,
          minLength: 1,
          maxLength: 30,
        },
        attribute: {},
      },
      legendClass: {
        title: 'Title ClassName',
        input: {
          type: 'text',
          value: '',
          required: true,
          disabled: false,
          minLength: 1,
          maxLength: 30,
        },
        attribute: {},
      },
      paddingTop: {
        title: 'Spacing Top',
        input: {
          type: 'number',
          value: 8,
          required: true,
          disabled: false,
          min: 0,
          max: 32,
        },
        attribute: {
          unit: 'px',
        },
      },
      paddingBottom: {
        title: 'Spacing Bottom',
        input: {
          type: 'number',
          value: 8,
          required: true,
          disabled: false,
          min: 0,
          max: 32,
        },
        attribute: {
          unit: 'px',
        },
      },
      paddingLeft: {
        title: 'Spacing Left',
        input: {
          type: 'number',
          value: 8,
          required: true,
          disabled: false,
          min: 0,
          max: 32,
        },
        attribute: {
          unit: 'px',
        },
      },
      paddingRight: {
        title: 'Spacing Right',
        input: {
          type: 'number',
          value: 8,
          required: true,
          disabled: false,
          min: 0,
          max: 32,
        },
        attribute: {
          unit: 'px',
        },
      },
    },
  },
  [ComponentPath.emailbox]: {
    title: 'Email Box',
    pathId: ComponentPath.emailbox,
    options: {
      id: {
        title: 'Input ID(#)',
        input: {
          type: 'text',
          value: 'id',
          required: true,
          disabled: false,
          maxLength: 100,
          minLength: 1,
        },
        attribute: {},
      },
      className: {
        title: 'Custom Class Name',
        input: {
          type: 'text',
          value: '',
          required: false,
          disabled: false,
          maxLength: 100,
          minLength: 1,
        },
        attribute: {},
      },
      label: {
        title: 'Title',
        input: {
          type: 'text',
          value: 'Label',
          required: true,
          disabled: false,
          maxLength: 100,
          minLength: 1,
        },
        attribute: {},
      },
      name: {
        title: 'Name',
        input: {
          type: 'text',
          value: 'name',
          required: true,
          disabled: false,
          maxLength: 100,
          minLength: 1,
        },
        attribute: {},
      },
      disabled: {
        title: 'Disabled',
        input: {
          type: 'checkbox',
          checked: false,
          name: 'isDisabled',
          value: 'disabled',
        },
        attribute: {},
      },
      required: {
        title: 'Required',
        input: {
          type: 'checkbox',
          checked: false,
          name: 'isRequired',
          value: 'required',
        },
        attribute: {},
      },
      placeholder: {
        title: 'Placeholder',
        input: {
          type: 'text',
          value: 'placeholder',
          required: true,
          disabled: false,
          maxLength: 1200,
          minLength: 1,
        },
        attribute: {},
      },
      pattern: {
        title: 'Pattern',
        input: {
          type: 'text',
          value: '',
          required: true,
          disabled: false,
          maxLength: 1200,
          minLength: 1,
        },
        attribute: {},
      },
      maxLength: {
        title: 'Max Length',
        input: {
          type: 'number',
          value: 0,
          required: false,
          disabled: false,
          min: 0,
        },
        attribute: {},
      },
      minLength: {
        title: 'Min Length',
        input: {
          type: 'number',
          value: 0,
          required: false,
          disabled: false,
          min: 0,
        },
        attribute: {},
      },
      invalidMessage: {
        title: 'Validation Message',
        input: {
          type: 'text',
          value: 'Current value is invalid',
          required: true,
          disabled: false,
          maxLength: 300,
          minLength: 1,
        },
        attribute: {},
      },
    },
  },
  [ComponentPath.radiobox]: {
    title: 'Radio Button',
    pathId: ComponentPath.radiobox,
    options: {
      id: {
        title: 'Input ID(#)',
        input: {
          type: 'text',
          value: 'id',
          required: true,
          disabled: false,
          maxLength: 100,
          minLength: 1,
        },
        attribute: {},
      },
      className: {
        title: 'Custom Class Name',
        input: {
          type: 'text',
          value: '',
          required: false,
          disabled: false,
          maxLength: 100,
          minLength: 1,
        },
        attribute: {},
      },
      label: {
        title: 'Label',
        input: {
          type: 'text',
          value: 'Label',
          required: true,
          disabled: false,
          maxLength: 100,
          minLength: 1,
        },
        attribute: {},
      },
      name: {
        title: 'Name',
        input: {
          type: 'text',
          value: 'name',
          required: true,
          disabled: false,
          maxLength: 100,
          minLength: 1,
        },
        attribute: {},
      },
      value: {
        title: 'Value',
        input: {
          type: 'text',
          value: '',
          required: true,
          disabled: false,
          maxLength: 250,
          minLength: 1,
        },
        attribute: {},
      },
      checked: {
        title: 'Selected',
        input: {
          type: 'checkbox',
          checked: false,
          name: 'isChecked',
          value: 'checked',
        },
        attribute: {},
      },
      disabled: {
        title: 'Disabled',
        input: {
          type: 'checkbox',
          checked: false,
          name: 'isDisabled',
          value: 'disabled',
        },
        attribute: {},
      },
      required: {
        title: 'Required',
        input: {
          type: 'checkbox',
          checked: false,
          name: 'isRequired',
          value: 'required',
        },
        attribute: {},
      },
      invalidMessage: {
        title: 'Validation Message',
        input: {
          type: 'text',
          value: 'Current value is invalid',
          required: true,
          disabled: false,
          maxLength: 300,
          minLength: 1,
        },
        attribute: {},
      },
    },
  },
  [ComponentPath.checkbox]: {
    title: 'Checkbox',
    pathId: ComponentPath.checkbox,
    options: {
      id: {
        title: 'Input ID(#)',
        input: {
          type: 'text',
          value: 'id',
          required: true,
          disabled: false,
          maxLength: 100,
          minLength: 1,
        },
        attribute: {},
      },
      className: {
        title: 'Custom Class Name',
        input: {
          type: 'text',
          value: '',
          required: false,
          disabled: false,
          maxLength: 100,
          minLength: 1,
        },
        attribute: {},
      },
      label: {
        title: 'Label',
        input: {
          type: 'text',
          value: 'Label',
          required: true,
          disabled: false,
          maxLength: 100,
          minLength: 1,
        },
        attribute: {},
      },
      name: {
        title: 'Name',
        input: {
          type: 'text',
          value: 'name',
          required: true,
          disabled: false,
          maxLength: 100,
          minLength: 1,
        },
        attribute: {},
      },
      value: {
        title: 'Value',
        input: {
          type: 'text',
          value: 'value',
          required: true,
          disabled: false,
          maxLength: 250,
          minLength: 1,
        },
        attribute: {},
      },
      checked: {
        title: 'Checked',
        input: {
          type: 'checkbox',
          checked: false,
          name: 'isChecked',
          value: 'checked',
        },
        attribute: {},
      },
      disabled: {
        title: 'Disabled',
        input: {
          type: 'checkbox',
          checked: false,
          name: 'isDisabled',
          value: 'disabled',
        },
        attribute: {},
      },
      required: {
        title: 'Required',
        input: {
          type: 'checkbox',
          checked: false,
          name: 'isRequired',
          value: 'required',
        },
        attribute: {},
      },
      invalidMessage: {
        title: 'Validation Message',
        input: {
          type: 'text',
          value: 'Current value is invalid',
          required: true,
          disabled: false,
          maxLength: 300,
          minLength: 1,
        },
        attribute: {},
      },
    },
  },
  [ComponentPath.selectbox]: {
    title: 'Select',
    pathId: ComponentPath.selectbox,
    options: {
      id: {
        title: 'Input ID(#)',
        input: {
          type: 'text',
          value: 'id',
          required: true,
          disabled: false,
          maxLength: 100,
          minLength: 1,
        },
        attribute: {},
      },
      className: {
        title: 'Custom Class Name',
        input: {
          type: 'text',
          value: '',
          required: false,
          disabled: false,
          maxLength: 100,
          minLength: 1,
        },
        attribute: {
          type: 'string',
          name: 'className',
        },
      },
      label: {
        title: 'Label',
        input: {
          type: 'text',
          value: 'Label',
          required: true,
          disabled: false,
          maxLength: 100,
          minLength: 1,
        },
        attribute: {},
      },
      name: {
        title: 'Name',
        input: {
          type: 'text',
          value: '',
          required: true,
          disabled: false,
          maxLength: 100,
          minLength: 1,
        },
        attribute: {},
      },
      disabled: {
        title: 'Disabled',
        input: {
          type: 'checkbox',
          checked: false,
          name: 'isDisabled',
          value: 'disabled',
        },
        attribute: {},
      },
      required: {
        title: 'Required',
        input: {
          type: 'checkbox',
          checked: false,
          name: 'isRequired',
          value: 'required',
        },
        attribute: {},
      },
      placeholder: {
        title: 'Placeholder',
        input: {
          type: 'text',
          value: '',
          required: true,
          disabled: false,
          maxLength: 1200,
          minLength: 1,
        },
        attribute: {},
      },
      options: {
        title: 'options',
        input: {
          type: 'table',
          optionItems: [{label: 'option', value: 'value'}],
          columns: {
            label: {
              title: 'Label',
              input: {
                type: 'text',
                value: '',
                name: 'label',
                required: true,
              },
            },
            value: {
              title: 'Value',
              input: {
                type: 'text',
                value: '',
                name: 'value',
                required: true,
              },
            },
          },
        },
        attribute: {},
      },
      invalidMessage: {
        title: 'Validation Message',
        input: {
          type: 'text',
          value: 'Current value is invalid',
          required: true,
          disabled: false,
          maxLength: 300,
          minLength: 1,
        },
        attribute: {},
      },
    },
  },
  [ComponentPath.numberbox]: {
    title: 'Number Box',
    pathId: ComponentPath.numberbox,
    options: {
      id: {
        title: 'Input ID(#)',
        input: {
          type: 'text',
          value: 'id',
          required: true,
          disabled: false,
          maxLength: 100,
          minLength: 1,
        },
        attribute: {},
      },
      className: {
        title: 'Custom Class Name',
        input: {
          type: 'text',
          value: '',
          required: false,
          disabled: false,
          maxLength: 100,
          minLength: 1,
        },
        attribute: {},
      },
      label: {
        title: 'Title',
        input: {
          type: 'text',
          value: 'Label',
          required: true,
          disabled: false,
          maxLength: 100,
          minLength: 1,
        },
        attribute: {},
      },
      name: {
        title: 'Name',
        input: {
          type: 'text',
          value: 'name',
          required: true,
          disabled: false,
          maxLength: 100,
          minLength: 1,
        },
        attribute: {},
      },
      disabled: {
        title: 'Disabled',
        input: {
          type: 'checkbox',
          checked: false,
          name: 'isDisabled',
          value: 'disabled',
        },
        attribute: {},
      },
      required: {
        title: 'Required',
        input: {
          type: 'checkbox',
          checked: false,
          name: 'isRequired',
          value: 'required',
        },
        attribute: {},
      },
      max: {
        title: 'Maximum',
        input: {
          type: 'number',
          value: 0,
          required: false,
          disabled: false,
          min: 0,
        },
        attribute: {},
      },
      min: {
        title: 'Minimum',
        input: {
          type: 'number',
          value: 0,
          required: false,
          disabled: false,
          min: 0,
        },
        attribute: {},
      },
      placeholder: {
        title: 'Placeholder',
        input: {
          type: 'text',
          value: '',
          required: true,
          disabled: false,
          maxLength: 1200,
          minLength: 1,
        },
        attribute: {},
      },
      invalidMessage: {
        title: 'Validation Message',
        input: {
          type: 'text',
          value: 'Current value is invalid',
          required: true,
          disabled: false,
          maxLength: 300,
          minLength: 1,
        },
        attribute: {},
      },
    },
  },
  [ComponentPath.passwordbox]: {
    title: 'Password',
    pathId: ComponentPath.passwordbox,
    options: {
      id: {
        title: 'Input ID(#)',
        input: {
          type: 'text',
          value: 'id',
          required: true,
          disabled: false,
          maxLength: 100,
          minLength: 1,
        },
        attribute: {},
      },
      className: {
        title: 'Custom Class Name',
        input: {
          type: 'text',
          value: '',
          required: false,
          disabled: false,
          maxLength: 100,
          minLength: 1,
        },
        attribute: {},
      },
      label: {
        title: 'Title',
        input: {
          type: 'text',
          value: 'Label',
          required: true,
          disabled: false,
          maxLength: 100,
          minLength: 1,
        },
        attribute: {},
      },
      name: {
        title: 'Name',
        input: {
          type: 'text',
          value: '',
          required: true,
          disabled: false,
          maxLength: 100,
          minLength: 1,
        },
        attribute: {},
      },
      disabled: {
        title: 'Disabled',
        input: {
          type: 'checkbox',
          checked: false,
          name: 'isDisabled',
          value: 'disabled',
        },
        attribute: {},
      },
      required: {
        title: 'Required',
        input: {
          type: 'checkbox',
          checked: false,
          name: 'isRequired',
          value: 'required',
        },
        attribute: {},
      },
      placeholder: {
        title: 'Placeholder',
        input: {
          type: 'text',
          value: '',
          required: true,
          disabled: false,
          maxLength: 1200,
          minLength: 1,
        },
        attribute: {},
      },
      pattern: {
        title: 'Pattern',
        input: {
          type: 'text',
          value: '',
          required: true,
          disabled: false,
          maxLength: 1200,
          minLength: 1,
        },
        attribute: {},
      },
      maxLength: {
        title: 'Max Length',
        input: {
          type: 'number',
          value: 0,
          required: false,
          disabled: false,
          min: 0,
        },
        attribute: {},
      },
      minLength: {
        title: 'Min Length',
        input: {
          type: 'number',
          value: 0,
          required: false,
          disabled: false,
          min: 0,
        },
        attribute: {},
      },
      invalidMessage: {
        title: 'Validation Message',
        input: {
          type: 'text',
          value: 'Current value is invalid',
          required: true,
          disabled: false,
          maxLength: 300,
          minLength: 1,
        },
        attribute: {},
      },
    },
  },
  [ComponentPath.datebox]: {
    title: 'Date Box',
    pathId: ComponentPath.datebox,
    options: {
      id: {
        title: 'Input ID(#)',
        input: {
          type: 'text',
          value: 'id',
          required: true,
          disabled: false,
          maxLength: 100,
          minLength: 1,
        },
        attribute: {},
      },
      className: {
        title: 'Custom Class Name',
        input: {
          type: 'text',
          value: '',
          required: false,
          disabled: false,
          maxLength: 100,
          minLength: 1,
        },
        attribute: {},
      },
      label: {
        title: 'Title',
        input: {
          type: 'text',
          value: 'Label',
          required: true,
          disabled: false,
          maxLength: 100,
          minLength: 1,
        },
        attribute: {},
      },
      name: {
        title: 'Name',
        input: {
          type: 'text',
          value: 'name',
          required: true,
          disabled: false,
          maxLength: 100,
          minLength: 1,
        },
        attribute: {},
      },
      disabled: {
        title: 'Disabled',
        input: {
          type: 'checkbox',
          checked: false,
          name: 'isDisabled',
          value: 'disabled',
        },
        attribute: {},
      },
      required: {
        title: 'Required',
        input: {
          type: 'checkbox',
          checked: false,
          name: 'isRequired',
          value: 'required',
        },
        attribute: {},
      },
      max: {
        title: 'Maximum',
        input: {
          type: 'date',
          value: '',
          required: true,
          disabled: false,
        },
        attribute: {},
      },
      min: {
        title: 'Minimum',
        input: {
          type: 'date',
          value: '',
          required: true,
          disabled: false,
        },
        attribute: {},
      },
      invalidMessage: {
        title: 'Validation Message',
        input: {
          type: 'text',
          value: 'Current value is invalid',
          required: true,
          disabled: false,
          maxLength: 300,
          minLength: 1,
        },
        attribute: {},
      },
    },
  },
  [ComponentPath.title]: {
    title: 'Title',
    pathId: ComponentPath.title,
    options: {
      paddingTop: {
        title: 'Spacing Top',
        input: {
          type: 'number',
          value: 8,
          required: true,
          disabled: false,
          min: 0,
          max: 32,
        },
        attribute: {
          unit: 'px',
        },
      },
      paddingBottom: {
        title: 'Spacing Bottom',
        input: {
          type: 'number',
          value: 8,
          required: true,
          disabled: false,
          min: 0,
          max: 32,
        },
        attribute: {
          unit: 'px',
        },
      },
      paddingLeft: {
        title: 'Spacing Left',
        input: {
          type: 'number',
          value: 8,
          required: true,
          disabled: false,
          min: 0,
          max: 32,
        },
        attribute: {
          unit: 'px',
        },
      },
      paddingRight: {
        title: 'Spacing Right',
        input: {
          type: 'number',
          value: 8,
          required: true,
          disabled: false,
          min: 0,
          max: 32,
        },
        attribute: {
          unit: 'px',
        },
      },
      id: {
        title: 'Input ID(#)',
        input: {
          type: 'text',
          value: 'ID',
          required: true,
          disabled: false,
          maxLength: 100,
          minLength: 1,
        },
        attribute: {},
      },
      className: {
        title: 'Custom Class Name',
        input: {
          type: 'text',
          value: '',
          required: false,
          disabled: false,
          maxLength: 100,
          minLength: 1,
        },
        attribute: {
          type: 'string',
          name: 'className',
        },
      },
      Value: {
        title: 'Value',
        input: {
          type: 'text',
          value: 'Title',
        },
        attribute: {},
      },
    },
  },
};
