// @steera/form-manager/dist/cjs/Comp/FormRenderer/RenderComponents/PasswordBox/PasswordBox.utils.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getValue = exports.getInvalidMessage = exports.getMinLength = exports.getMaxLength = exports.getPattern = exports.getClassName = exports.getPlaceholder = exports.getRequired = exports.getDisabled = exports.getName = exports.getLabel = exports.getID = void 0;
function getID(option) {
    return option.input.value;
}
exports.getID = getID;
function getLabel(option) {
    return option.input.value;
}
exports.getLabel = getLabel;
function getName(option) {
    return option.input.value;
}
exports.getName = getName;
function getDisabled(option) {
    return option.input.checked;
}
exports.getDisabled = getDisabled;
function getRequired(option) {
    return option.input.checked;
}
exports.getRequired = getRequired;
function getPlaceholder(option) {
    return option.input.value;
}
exports.getPlaceholder = getPlaceholder;
function getClassName(option) {
    return option.input.value;
}
exports.getClassName = getClassName;
function getPattern(option) {
    return option.input.value;
}
exports.getPattern = getPattern;
function getMaxLength(option) {
    if (option.input.value === 0 || option.input.value < 0)
        return undefined;
    else
        return option.input.value;
}
exports.getMaxLength = getMaxLength;
function getMinLength(option) {
    if (option.input.value === 0 || option.input.value < 0)
        return undefined;
    else
        return option.input.value;
}
exports.getMinLength = getMinLength;
function getInvalidMessage(option) {
    return option.input.value;
}
exports.getInvalidMessage = getInvalidMessage;
function getValue(templateValues, name) {
    if (templateValues && templateValues.hasOwnProperty(name)) {
        return templateValues[name];
    }
}
exports.getValue = getValue;
