// @steera/form-manager/dist/cjs/Comp/FormRenderer/RenderComponents/CheckBox/CheckBox.utils.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getInvalidMessage = exports.getClassName = exports.getChecked = exports.getRequired = exports.getDisabled = exports.getValue = exports.getName = exports.getLabel = exports.getID = void 0;
function getID(option) {
    return option.input.value;
}
exports.getID = getID;
function getLabel(option) {
    return option.input.value;
}
exports.getLabel = getLabel;
function getName(option) {
    return option.input.value;
}
exports.getName = getName;
function getValue(option) {
    return option.input.value;
}
exports.getValue = getValue;
function getDisabled(option) {
    return option.input.checked;
}
exports.getDisabled = getDisabled;
function getRequired(option) {
    return option.input.checked;
}
exports.getRequired = getRequired;
function getChecked(option) {
    return option.input.checked;
}
exports.getChecked = getChecked;
function getClassName(option) {
    return option.input.value;
}
exports.getClassName = getClassName;
function getInvalidMessage(option) {
    return option.input.value;
}
exports.getInvalidMessage = getInvalidMessage;
