// @steera/form-manager/dist/cjs/Comp/FormGenerator/RenderComponents/SelectBox/SelectBox.utils.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getOptions = exports.getInvalidMessage = exports.getClassName = exports.getPlaceholder = exports.getRequired = exports.getDisabled = exports.getName = exports.getLabel = exports.getID = void 0;
function getID(option) {
    return option.input.value;
}
exports.getID = getID;
function getLabel(option) {
    return option.input.value;
}
exports.getLabel = getLabel;
function getName(option) {
    return option.input.value;
}
exports.getName = getName;
function getDisabled(option) {
    return option.input.checked;
}
exports.getDisabled = getDisabled;
function getRequired(option) {
    return option.input.checked;
}
exports.getRequired = getRequired;
function getPlaceholder(option) {
    return option.input.value;
}
exports.getPlaceholder = getPlaceholder;
function getClassName(option) {
    return option.input.value;
}
exports.getClassName = getClassName;
function getInvalidMessage(option) {
    return option.input.value;
}
exports.getInvalidMessage = getInvalidMessage;
function getOptions(option) {
    return option.input.optionItems;
}
exports.getOptions = getOptions;
